### PR TITLE
Face lift

### PR DESCRIPTION
**Pull request by hesh64**

### CODE REVIEW

File: src/couch2/index.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        }
+
+        // didn't match so undefined
+        return undefined;
+      }
+    }) as typeof ctor;
+  };
+}
+
+@KeyGen(() => {
```
Comment (hesh64) [inline src/couch2/index.ts:291]: Everything from here down needs to be deleted

File: src/couch2/index.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+const col = bucket.defaultCollection();
+
+// col.getAndLock('airline_10', 1000).then(console.log).catch(console.log)
+
+// console.log(bucket)
+// cluster._connect();
+
+const cache = {};
+
+function MakeResponseProxy() {
```
Comment (hesh64) [inline src/couch2/index.ts:17]: this still needs more work.

File: src/couch2/index.ts [hunk 3]
```diff
@@ -0,0 +1,393 @@
+const cb = require('couchbase');
```
Comment (hesh64) [inline src/couch2/index.ts:1]: remove this

File: src/couch2/index.ts [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+});
+
+const bucket = cluster.bucket('travel-sample');
+const col = bucket.defaultCollection();
+
+// col.getAndLock('airline_10', 1000).then(console.log).catch(console.log)
+
+// console.log(bucket)
+// cluster._connect();
+
```
Comment (hesh64) [inline src/couch2/index.ts:14]: until here

File: src/couch2/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      opArrays[arrayIdx] = [];
+    }
+    opArrays[arrayIdx].push(op);
+  });
+  return opArrays;
+};
+
+export const slice16 = (ops: Array<any>) => slicen(ops, 16);
+
+export const chopRequest = (m, id: string, fnName: 'mutateIn' | 'lookupIn', ops) => {
```
Comment (ryan-barrett) [inline src/couch2/helpers.ts:16]: nice

**Pull request merged**